Assets/Scripts/Effect.cs

```diff
@@ -22,9 +22,9 @@ void Update()
         
     }
 
-   public void Action()
+   public void Action(Vector2 cursorPos)
```

Comment:
Refactor this variable name to simply be position. In the context of the "Action" method, all it cares about is that it knows the position to do the action, it does not care where the position came from. It could be a cursor or something else like say a unit or environmental trigger.

Of course, when Action is called it will often be passed the cursor position, but once it's inside action it is just the position where the action will happen.